glean-core/preview/src/core_metrics.rs

```diff
@@ -0,0 +1,113 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use glean_core::{metrics::StringMetric, CommonMetricData, Lifetime};
+
+/// Metrics included in every ping as `client_info`.
+#[derive(Debug)]
+pub struct ClientInfoMetrics {
+    /// The build identifier generated by the CI system (e.g. "1234/A").
+    pub app_build: String,
+    /// The user visible version string (e.g. "1.0.3").
+    pub app_display_version: String,
+    /// The channel the application is being distributed on.
+    pub app_channel: Option<String>,
+    /// The name of the operating system.
+    pub os: String,
```

Comment:
Can this be an internal implementation detail computed with `#[cfg(target_os = "windows")]`?

Reply:
We could, and if we wanted to support all the targets Rust supports we could use https://docs.rs/platforms/0.2.1/platforms/target/enum.OS.html to make it easy.

Reply:
> 
> 
> We could, and if we wanted to support all the targets Rust supports we could use https://docs.rs/platforms/0.2.1/platforms/target/enum.OS.html to make it easy.

If we can use the RUST APIs on all the Rust targets, then this is the way to go, IMHO :)

Reply:
If we wanted we could certainly move detection of that field into glean-core.

Reply:
(putting this into glean-preview _for now_ would allow us to update m-c before the holidays and only cut a proper glean release when we're all back)

Reply:
Yes, it's fine to put this in glean-preview and we can then figure out if/when to move to glean-core. I have a slight preference toward stubbing out things vs providing new API surface (even if it needs to be scraped in the future), but I have no strong opinion on this. Your call!